### DISCUSSION TITLE
Question on State Space simulation in nodal coordinate with ROM and controllability of the State Space system

### DISCUSSION THREAD

**Discussion by xxyao-95** [ImperialCollegeLondon/sharpy#156] [Q&A] [answered]
Hi
I am following the discussion in https://github.com/ImperialCollegeLondon/sharpy/discussions/152. I extracted the state space system from goland wing geometry in nodal coordinate as discussed. I used 8 spanwise discretisation and hence there are 9 nodes on the beam. The extracted B matrix has dimensions 36 x 147 and D matrix has dimensions 144 x 147. It shows the input vector u has 147 element and output vector y has 144 element. This makes great sense as u has 3 more entries than y, which are gust velocity, control surface deflection and its velocity. Although, the input/output vector makes sense, I have some problems in running simulations with the statespace system, and I got some questions that I hope you could help me with.

1) Could you confirm that u vector is stricly [eta, eta_dot, w_g, delta, delta_dot, Q_ext] ? In my understanding,  eta, eta_dot and Q_ext are vectors with (no. node -1) *6 entries (6 degrees of freedom), and w_g, delta and delta_dot are scalars. Hence with 9 nodes, eta, eta_dot and Q_ext takes totally (9-1)*6*3 = 144 entries and 144 + 3 scalars gives u vector 147 entires. Therefore, the gust input should be at the 97th entry if u vector is as stated. Could you confirm that my assumption is correct?

2) The control surface input shoud be at the 98 and 99th entry if u vector is as stated. The physical system of goland wing should be controllable with control surface input. Hence if I reduce the B matrix to only include the columns for control surface input, the state system should still be controllable. However, the rank of controllability matrix is 2 when I calculate with A and reduced B matrix. Hence is there something wrong with the state space matrices I extracted?

3) For the output vector y, could you confirm that the sequence is [Q, eta, eta_dot]? I want observe the tip deflection of the wing under turbulence and I need to extract the result from y vector. I counted in paraview that the node number is 4 and 5 on the tip, but I am not very sure how data is organised in the y vector?

**Comment by ngoiz** [comment, marked as answer]
Yes, it looks like you have the vector ordering nailed down. u_gust at the 97th entry, delta at 98th and delta_dot at 99th. I usually refer to Q as the modal forces but I think in your case you are referring to nodal forces (again size 6*(n_node - 1)). Your output is correct too [N, eta, eta_dot]. For each node, its 6DOF vector is `R_x, R_y, R_z, Psi_x, Psi_y, Psi_z` (i.e. the position and cartesian rotation vector) in A frame. Thus you need to count in the output for the size of `N` + 6 * 3 nodes (i.e this gets you onto the vector of node 4) and then your indices of interest.

As for the controllability I cannot really say...maybe some states aren't controllable via the control surface deflection input, for instance the gust states... but don't take my word on that.

**Reply by xxyao-95** [reply]
Hi,
For the controllability part, I noticed that with M = 2 and N = 8, at U_inf = 50, sharpy would give a warning as follows:
UserWarning: Reduced Order Model Unstable
  warn.warn('Reduced Order Model Unstable')

I tried to play around with the number of discretizations and I realize that when there is no warning, the system is controllable by  2 inputs, and when ROM is unstable the system is not controllable by only 2 input.

Could you elaborate a bit more on what makes ROM unstable?

**Reply by ngoiz** [reply]
ROM methods work in several manners. Krylov ROMs work by matching transfer functions but stability is not guaranteed. I have seen that with refined grids it tends to be stable, if unstable increase `r` and that might fix it. But it's part of the method though, one of its drawbacks.

**Reply by xxyao-95** [reply]
Thanks for the explaination!
